### ISSUE TITLE
CsvWriter calls .Flush() after every line?

### ISSUE THREAD

**Issue by rdingwall** [JoshClose/CsvHelper#38]
Hi, we're writing a big CSV to a network share (134 columns, 100k rows, 50MB on disk) and the performance is really slow compared to the local file system (much slower than expected... we have a gige lan and a big SAN, it's normally pretty quick copying big files around). 

Digging around a bit, I notice CsvWriter.NextRecord() calls TextWriter.Flush() after every record.
1. This seems a little bit excessive in terms of IO... wouldn't it be better to never call .Flush() and let the FileStream automatically decide when to flush itself? (Every 4KB by default I believe)
2. Because of the manual .Flush() calls it is currently not possible for me to configure how often CsvHelper writes to disk, even if I pass my own buffered stream in. If you let FileStream do the flushing automatically, I could specify my buffer size in the stream constructor (say 500KB) and it would automatically flush every time this limit was met.

Sorry I don't have a pull request for this, I'm at work.

**Comment by mevatron** [comment]
@rdingwall Assuming that you're working over an SMB share, I believe you've hit the exact same issue we're seeing with our programs and NAS. I've posted a detailed write-up of what I've learned on [ServerFault][1]. It appears from my testing that Windows automatically flushes writes to any network share (I've tried both NFS and SMB and Windows behaves the same). 

Did you end up solving it with an internal buffering system then? It looks like that's what we will have to do as well for our Windows clients.

[1]: https://serverfault.com/questions/808633/small-writes-to-smb-network-share-are-slow-on-windows-fast-over-cifs-linux-moun
